### DISCUSSION TITLE
Troubleshooting when not all movies are found

### DISCUSSION THREAD

**Discussion by ThomGreenie** [mza921/Plex-Auto-Collections#185] [Q&A] [answered]
Hello everyone!

First, thanks very much for this script. It's a godsend to not have to go through filling out collections manually. I've set up everything I want (TMDB, Trakt, Plex) but for some of the lists I put in, the script does not find all movies even though I have them in my library. Does anyone know a solution for this? Do I have some erroneous settings somewhere? I've tried cleaning up my library and checking whether or not the files are named oddly but they're not.

I would appreciate any help with this.

**Comment by tehniemer** [comment]
Has PLEX matched everything in your library? Unmatched files can be a problem.

**Reply by ThomGreenie** [reply]
Thanks for your response. Plex matches everything in the library. I think the problem is in my metadata agent. I've noticed how the script doesn't find movies in the TMDB database if the movie's TMDB rating doesn't show in Plex. 

So far I've tried different agents and I can never get them to show up consistently.

**Reply by ramair02** [reply]
Metadata agents are also configurable per Library. Check from the Plex/Web homepage by clicking the three dots menu next to a Library > Manage Library > Edit. Go to Advanced. You probably want the new agent, so make sure "Agent" is set to "Plex Movie". Then you can close that window and again go to the three dots menu > Manage Library > Refresh All Metadata. Do this for each Library.

I mention this since your Libraries may be configured differently than your global Agent selections. Maybe it will help.

**Comment by burkasaurusrex** [comment, marked as answer]
Try `Manage Library` > `Refresh All Metadata`. Especially if you're using the new metadata matching agent, sometimes it takes a couple of refreshes.

**Reply by ThomGreenie** [reply]
Thank you! This was it. After I ran the refresh metadata again it worked like a charm.
